Calculator/Functions.cpp

```diff
@@ -53,252 +77,419 @@ bool is_declared(string var)
     return false;
 }
 
-double define_name(string var, double val)
+//------------------------------------------------------------------------------
+// Calls function "is_declared" passing the argument "string var" to check if 
+// argument "string var" has already been declared (returns true). If it has been 
+// declared an error is thrown. If argument "string var" has NOT been previously declared
+// (returns false), it is then added to vector "var_table" along with the value property argument
+//  "double val" and value property "double val" is returned
+
+double define_name(string var, double val, vector<Variable>& var_table)
 {
-    if (is_declared(var)) {
-        error(var, " declared twice");
+    if (is_declared(var, var_table)) {
+        error(var, " previously declared; use 'reset' to change the value.");
     }
     var_table.push_back(Variable{ var, val });
     return val;
 }
 
-double squareroot()
+//------------------------------------------------------------------------------
+// "double c" is set to the result of function "expression()", Token t is set to
+// the next token in Token_stream 'ts', which is returned by  function "ts.get()"
+// If the kind property of the token is not a ')' an error is thrown
+// "double c" is returned
+
+double parenthesis(Token_stream& myts, vector<Variable>& var_table)
+{
+    double c = expression(myts, var_table);
+    Token t = myts.get();
+    if (t.kind != ')') {
```

Comment:
move all declarations to the top and put a space between declarations and algorithm